src/fu-device-list.c

```diff
@@ -569,9 +569,12 @@ fu_device_list_add (FuDeviceList *self, FuDevice *device)
 		return;
 	}
 
-	/* added the same device from a different plugin */
-	if (item != NULL && g_strcmp0 (fu_device_get_plugin (item->device),
-				       fu_device_get_plugin (device)) != 0) {
+	/* added the same device, supporting same protocol, from a different plugin */
+	if (item != NULL &&
+	    g_strcmp0 (fu_device_get_plugin (item->device),
+		       fu_device_get_plugin (device)) != 0 &&
+	    g_strcmp0 (fu_device_get_protocol (item->device),
+		       fu_device_get_protocol (device)) == 0) {
```

Comment:
So just a thought; we added this chunk to dedupe things like the TPM (Dell upgradable, fallback one not so) -- isn't this going to break that dedupe action?

Reply:
The original reason it was added was Synaptics MST flashing, so that the `dell-dock` plugin got preferred over the `synapticsmst` plugin for resiliency, although I think it gets used for the thunderbolt/dell-dock as well.  Let me double check all 3 of those aren't ruined by this.

Reply:
If it helps, I'm not against a .quirk for ALLOW_DUPLICATE_GUIDS to disable this check. Putting that on UEFI **or** NVMe would be quite a nice workaround.

Reply:
I don't have the HDD scenario handy anymore, but I did check it before I submitted the PR.

TBT is ok because there is an explicit check for whether you're connected a thunderbolt port now in the code.  The dedupe code doesn't run:
https://github.com/fwupd/fwupd/blob/9a68084dbc2c5886c927875f977e8f50f9161bc3/plugins/dell-dock/fu-dell-dock-i2c-ec.c#L185

Synaptics MST is ok because of the priority check matches identically in both plugins:

```
20:26:09:0011 FuDeviceList         ignoring device 585c346c009da3cc19a400fb59a4ed8d0cbd0151 [synaptics_mst] as better device 0060e00540acb5ded8f04c214fb157816bbbb6a6 [dell_dock] already exists
```

TPM is ok, but only because it hits an earlier check.
```
20:30:07:0884 FuDeviceList         device c6a80ac3a22083423992a3cb15018989f37834d6 already exists, ignoring
```

So it does work
```
$ sudo fwupdtool get-devices --plugin-whitelist=dell --plugin-whitelist=tpm --plugin-whitelist=uefi
Loading…                 [***************************************]
XPS 13 9380
│
├─System Firmware:
│     Device ID:           6c24a747f97668873b761558e322398a91dbf394
│     Current version:     1.9.1
│     Minimum Version:     1.9.1
│     Vendor:              Dell Inc. (DMI:Dell Inc.)
│     GUID:                ce945437-7358-49f1-95d8-6b694a10a755
│     Device Flags:        • Internal device
│                          • Updatable
│                          • Requires AC power
│                          • Needs a reboot after installation
│                          • Cryptographic hash verification is available
│                          • Device is usable for the duration of the update
│   
└─TPM 2.0:
      Device ID:           c6a80ac3a22083423992a3cb15018989f37834d6
      Summary:             Platform TPM device
      Current version:     7.2.0.2
      Vendor:              Dell Inc. (PCI:0x1028)
      Update Error:        Updating disabled due to TPM ownership
      GUIDs:               d621e687-b00d-539a-aeac-ac3eb530cbc5 ← 08af-2.0
                           ff71992e-52f7-5eea-94ef-883e56e034c6 ← system-tpm
                           7d65b10b-bb24-552d-ade5-590b3b278188 ← DELL-TPM-2.0-NTC-NPCT
                           6f5ddd3a-8339-5b2a-b9a6-cf3b92f6c86d ← DELL-TPM-2.0-NTC-NPCT75x
                           fe462d4a-e48f-5069-9172-47330fc5e838 ← DELL-TPM-2.0-NTC-NPCT75xrls
      Device Flags:        • Internal device
                           • Requires AC power
```

Reply:
> If it helps, I'm not against a .quirk for ALLOW_DUPLICATE_GUIDS to disable this check. Putting that on UEFI **or** NVMe would be quite a nice workaround.

I wouldn't rule out that this scenario happen in the future with a different device, so I would prefer a generic solution if we can keep one.  If this breaks something else in the future and we need to revert it a quirk for uefi/nvme will be fine too.